### ISSUE TITLE
Comparison of GB & loss policies

### ISSUE THREAD

**Issue by arogozhnikov** [arogozhnikov/hep_ml#18]
Probably some test notebook would be helpful to see how changes in losses affect training process.

**Comment by arogozhnikov** [comment]
Check on some weighted dataset several options:
* sklearn (logloss, exploss)
* xgboost (logloss)
* UGBoost 
  * logloss, exploss only with gradient * weight as target 
  * logloss, exploss with target = gradient, weight=weight
  * logloss, exploss with target = gradient/hessian, weight = hessian

**Comment by arogozhnikov** [comment]
todo: find out why there is such difference in variations of predictions

**Comment by arogozhnikov** [comment]
Some experiments I've done long ago say that:
- UGradientBoosting update policy (all data  + regularization) fails on the start, but turns out to be more stable when hundreds or thousands of trees are used.
- see also discussion on [kaggle](https://www.kaggle.com/c/flavours-of-physics/forums/t/17008/sharing-code-ugbc), which confirms this point.

As of last point (setting target and weight), there seems to be no strong winner. 
Mostly following tradition, I'll prefer grad/hess version.
